Четвёртая лабораторная работа выполнена

PopovMatvey opened the pull request:
1) Были выполнены все задания 1-14;
2) Были выполнены все линтеры в заданиях 1-14;
3) Были выполнены все тесты в заданиях 1-14.

PopovMatvey committed:
Четвёртая лабораторная работа выполнена

1) Были выполнены все задания с 1-го по 14-й;
2) Были выполнены все линтеры в заданиях с 1-го по 14-й;
3) Били выполнены все тесты в заданиях с 1-го по 14-й.

djilua1 commented on src/ex4_js-objects-part2/task-01.js:
```diff
@@ -0,0 +1,15 @@
+  });
+
+  return returnedValue;
+}
```
Слишком сложно, можно гораздо легче + много лишнего.

djilua1 commented on src/ex4_js-objects-part2/task-03.js:
```diff
@@ -0,0 +1,9 @@
+function deleteFirstAndLastSpace(string) {
+  if ((string[0] === ' ') && (string[string.length - 1] === ' ')) {
```
А если первый пробле, а последний нет?

djilua1 commented on src/ex4_js-objects-part2/task-05.js:
```diff
@@ -0,0 +1,9 @@
+  }
+
+  return false;
+}
```
`return string.indexOf(substring)) !== -1;`

djilua1 commented on src/ex4_js-objects-part2/task-07.js:
```diff
@@ -0,0 +1,15 @@
+  }
+
+  return returnedString;
+}
```
Можно проще. ПРосто сравнить длинну с переменной.

djilua1 commented on src/ex4_js-objects-part2/task-08.js:
```diff
@@ -0,0 +1,15 @@
+  }
+
+  return returnedString[0].toLowerCase() + returnedString.substring(1);
+}
```
split + join.

djilua1 commented on src/ex4_js-objects-part2/task-09.js:
```diff
@@ -0,0 +1,18 @@
+      }
+    }
+    concatenationString += string[i];
+  }
```
Не явное решение, его тяжело понять. Давай сделаем попонятнее, через массивы например.

djilua1 commented on src/ex4_js-objects-part2/task-10.js:
```diff
@@ -0,0 +1,11 @@
+  }
+
+  return reversString;
+}
```
Есть специальный метод.

djilua1 commented on src/ex4_js-objects-part2/task-11.js:
```diff
@@ -0,0 +1,25 @@
+    if (!object.hasOwnProperty(string[i])) {
+      object[string[i]] = 0;
+    }
+    if (object.hasOwnProperty(string[i])) {
```
Отбить пустой строкой.

djilua1 commented on src/ex4_js-objects-part2/task-13.js:
```diff
@@ -0,0 +1,5 @@
+function getValueInRange() {
+  return Math.random() * 100;
```
Никогда не получим 100.

PopovMatvey committed:
Исправления по четвёртой лабораторной работе

Были исправлены все девять замечаний:
1) Был упрощён алгоритм, также было убрано всё лишнее в первом задании;
2) Были добавлены проверки на наличие пробела только в начале или только в конце в третьем задании; 
3) Был оптимизирован return в вид: "return string.indexOf(substring)) !== -1;";
4) Был упрощён алгоритм в седьмом задании;
5) return был изменён с return returnedString[0].toLowerCase() + returnedString.substring(1);
 на return returnedString.split(/^/).map((word) => word[0].toLowerCase() + word.substring(1)).join(''); в восьмом задании;
6) Решение в девятом задании стало более явным, для реализации использовал массивы;
7) Для решения десятого задания был использован специальный метод;
8) В методе getObjectWithAmountChars одиннадцатого задания условия были отбиты пустой строкой;
9) В тринадцатом задании результат выполнения функции теперь может нам выдать 100.
